### DISCUSSION TITLE
output.json -> output.ndjson

### DISCUSSION THREAD

**Discussion by nicpenning** [blacklanternsecurity/bbot#734] [Ideas]
It seems that the output.json file of a BBOT scan is actually new line delimited JSON. Would it then be appropriate to rename the output.json file to output.ndjson?

**Comment by TheTechromancer** [comment]
Not a bad idea. Would you be willing to make a PR? There are probably a few places in the tests and docs it would also need to be updated.

**Reply by nicpenning** [reply]
I will give it a go. I am new to the project so I will see what I can do. Thanks!

**Reply by TheTechromancer** [reply]
Sounds good, feel free to open it early and I can help out along the way if needed.

**Comment by nicpenning** [comment]
I would like to try using poetry and develop BBOT, but the steps here did not work: https://www.blacklanternsecurity.com/bbot/contribution/

After properly installing poetry, adding it to the path, and then running `poetry shell`, I could not execute `bbot --help` as it does not appear to be a command I can execute yet.

```BASH
user@system-1:~/bbot$ poetry shell
Spawning shell within /home/user/.cache/pypoetry/virtualenvs/bbot-Kav3PRTN-py3.10
user@system-1:~/bbot$ . /home/user/.cache/pypoetry/virtualenvs/bbot-Kav3PRTN-py3.10/bin/activate
(bbot-py3.10) user@system-1:~/bbot$ bbot --help
Command 'bbot' not found, did you mean:
  command 'bbox' from deb ps2eps (1.68+binaryfree-3)
  command 'bnbot' from deb pvpgn (1.8.5-2.1ubuntu1)
  command 'brot' from deb fortunes-de (0.34-1)
Try: sudo apt install <deb name>
```

**Reply by TheTechromancer** [reply]
I believe you have to do `poetry install` as well.

**Reply by nicpenning** [reply]
You have to do poetry install twice?

I followed these instructions:
```
# clone your forked repo and cd into it
git clone git@github.com/<username>/bbot.git && cd bbot

# install poetry
curl -sSL https://install.python-poetry.org | python3 -

# install pip dependencies
poetry install

# enter virtual environment
poetry shell

bbot --help
```

I am unsure how `bbot` is a runnable command using the raw code and not a container or other method. 

I will try and dig deeper to see how to run my own copy of bbot.

**Reply by TheTechromancer** [reply]
A critical step that's easy to miss is being inside the `bbot` folder (`&& cd bbot`) before running `poetry install`. I've done that on multiple occasions.

**Reply by nicpenning** [reply]
I will try that, but I was pretty sure I followed those steps verbatim. I will report back.

**Comment by nicpenning** [comment]
I re-ran the poetry install in the bbot directory and it seems that I am making progress!

**Comment by liquidsec** [comment]
we dont use poetry anymore
